functors task (1)/student/src/impl.cpp

```diff
@@ -1,2 +1,17 @@
 #include "impl.h"
 
+BeerBrand BeerOrganizer::operator()() const
+{
+    if((counter + 1) >= static_cast<int8_t>(BeerBrand::Max))
+    {
+        counter = static_cast<int8_t>(BeerBrand::None);
+    }
+    return static_cast<BeerBrand>(++counter);
+}
+
+std::map <std::pair<AlcoholDrink, NonAlcoholDrink>, Cocktail> MixingPolicy::bar =
```

Comment:
Mapping is the best solution here, so you're right.
It is more flexible switch logic.